src/interfaces/IMessageDispatcherArbitrum.sol

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.16;
+
+import { MessageLib, IMessageDispatcher, IBatchMessageDispatcher } from "./IBatchMessageDispatcher.sol";
+
+/**
+ * @title ERC-5164: Cross-Chain Execution Standard
+ * @dev IMessageDispatcher interface extended to support Arbitrum two steps message dispatch.
+ * @dev See https://eips.ethereum.org/EIPS/eip-5164
+ */
+interface IMessageDispatcherArbitrum is IBatchMessageDispatcher {
+  /**
+   * @notice Emitted once a message has been processed and put in the Arbitrum inbox.
+   * @dev Using the `ticketId`, this message can be reexecuted for some fixed amount of time if it reverts.
+   * @param messageId ID uniquely identifying the message
+   * @param sender Address who processed the message
+   * @param ticketId ID of the newly created retryable ticket
+   */
+  event MessageProcessed(
+    bytes32 indexed messageId,
+    address indexed sender,
+    uint256 indexed ticketId
+  );
+
+  /**
+   * @notice Emitted once a message has been processed and put in the Arbitrum inbox.
+   * @dev Using the `ticketId`, this message can be reexecuted for some fixed amount of time if it reverts.
+   * @param messageId ID uniquely identifying the messages
+   * @param sender Address who processed the messages
+   * @param ticketId ID of the newly created retryable ticket
+   */
+  event MessageBatchProcessed(
+    bytes32 indexed messageId,
+    address indexed sender,
+    uint256 indexed ticketId
+  );
+
+  /**
+   * @notice Emitted when a message has successfully been dispatched and put in the Arbitrum inbox.
+   * @param messageId ID uniquely identifying the message
+   * @param from Address that dispatched the message
+   * @param toChainId ID of the chain receiving the message
+   * @param to Address that will receive the message
+   * @param data Data that was dispatched
+   * @param ticketId ID of the newly created retryable ticket
+   */
+  event MessageDispatchedAndProcessed(
+    bytes32 indexed messageId,
+    address indexed from,
+    uint256 toChainId,
+    address to,
+    bytes data,
+    uint256 indexed ticketId
+  );
+
+  /**
+   * @notice Emitted when a batch of messages has successfully been dispatched and put in the Arbitrum inbox.
+   * @param messageId ID uniquely identifying the batch of messages
+   * @param from Address that dispatched the batch of messages
+   * @param toChainId ID of the chain receiving the batch of messages
+   * @param messages Array of Message that was dispatched
+   * @param ticketId ID of the newly created retryable ticket
+   */
+  event MessageBatchDispatchedAndProcessed(
+    bytes32 indexed messageId,
+    address indexed from,
+    uint256 toChainId,
+    MessageLib.Message[] messages,
+    uint256 indexed ticketId
+  );
```

Comment:
We already have the `MessageDispatched`, `MessageProcessed`, `MessageBatchDispatched`, and `MessageBatchProcessed` events so I think we should use those instead of creating these two new events (`MessageDispatchedAndProcessed`, `MessageBatchDispatchedAndProcessed`). They have all the same information and it would make things easier for people doing event queries.

Reply:
Fixed in this commit: https://github.com/GenerationSoftware/ERC5164/pull/4/commits/5d324546a659754db585150ad25e667018f8f878